src/main/java/slimeknights/tconstruct/gadgets/GadgetClientEvents.java

```diff
@@ -29,8 +29,8 @@
   @SubscribeEvent
   public static void registerItemColors(ColorHandlerEvent.Item event) {
     final ItemColors colors = event.getItemColors();
-    colors.register((@Nonnull ItemStack stack, int tintIndex) -> SlimeSlingItem.getColorFromStack(stack), GadgetItems.slime_sling_blue, GadgetItems.slime_sling_purple, GadgetItems.slime_sling_magma, GadgetItems.slime_sling_green, GadgetItems.slime_sling_blood);
-    colors.register((@Nonnull ItemStack stack, int tintIndex) -> SlimeBootsItem.getColorFromStack(stack), GadgetItems.slime_boots_blue, GadgetItems.slime_boots_purple, GadgetItems.slime_boots_magma, GadgetItems.slime_boots_green, GadgetItems.slime_boots_blood);
+    colors.register((@Nonnull ItemStack stack, int tintIndex) -> SlimeSlingItem.getColorFromStack(stack), GadgetItems.slime_sling_blue.get(), GadgetItems.slime_sling_purple.get(), GadgetItems.slime_sling_magma.get(), GadgetItems.slime_sling_green.get(), GadgetItems.slime_sling_blood.get());
+    colors.register((@Nonnull ItemStack stack, int tintIndex) -> SlimeBootsItem.getColorFromStack(stack), GadgetItems.slime_boots_blue.get(), GadgetItems.slime_boots_purple.get(), GadgetItems.slime_boots_magma.get(), GadgetItems.slime_boots_green.get(), GadgetItems.slime_boots_blood.get());
```

Comment:
This method is so much of a mess, and its calling a messy function. To clean it up, use `registerEnum` on slime sling and slime boots, then change the code to teh following:
```java
for (SlimeType slime : SlimeType.values()) {
  colors.register((stack,index)->slime.getBallColor(), GadgetItems.slime_boots.get(slime), GadgetItems.slime_sling.get(slime));
}
```
And you can then delete the messy methods from slime sling and slime boots

Reply:
Nifty. Will do so.